compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/DesktopPopup.desktop.kt

```diff
@@ -44,15 +42,8 @@ fun rememberCursorPositionProvider(
     alignment: Alignment = Alignment.BottomEnd,
     windowMargin: Dp = 4.dp
 ): PopupPositionProvider = with(LocalDensity.current) {
-    val component = LocalLayerContainer.current
-    val cursorPoint = remember {
-        val awtMousePosition = MouseInfo.getPointerInfo().location
-        convertPointFromScreen(awtMousePosition, component)
-        IntOffset(
-            (awtMousePosition.x * component.density.density).toInt(),
-            (awtMousePosition.y * component.density.density).toInt()
-        )
-    }
+    val pointerPositions = LocalPointerPositions.current
+    val cursorPoint = remember { pointerPositions.entries.first().value.round() }
```

Comment:
Shouldn't this be
1. `rememberUpdatedState`
2. `firstOrNull` (for example, I want to show a popup in a test; there is no mouse position, and this crashes)

Reply:
1. It shouldn't because it should be initialized once and not changed in recompositions
2. It shouldn't be `firstOrNull` as well, I guess (or we should return nullable PositionProvider, it will break compatibility). We cannot create "default" position for this case, since it will cause weird behavior. For tests this error is an indication that a pointer event hasn't been sent.

Reply:
In any case, a minor improvement:
```
pointerPositions.values.first()
```

Reply:
I've implemented behaviour: if there is no mouse or mouse is outside of the scene -- popup will be shown under the bottom-center of anchor component.
I don't want to document it, so it will be kinda undefined behaviour of "related to cursor" statement

Reply:
> I've implemented behaviour: if there is no mouse or mouse is outside of the scene -- popup will be shown under the bottom-center of anchor component.

If the component fills the window, does that mean the popup won't be visible?

Reply:
It will be visible, we align popup position according to component position and window bounds